mopub-sdk/mopub-sdk-base/src/main/java/com/mopub/network/AdRequest.java

```diff
@@ -403,7 +406,21 @@ public static void setServerOverrideListener(
             }
         }
 
-        builder.setRawPayload(new JSONObjectSerializable(jsonHeaders));
+        String adSourceId = null;
+        try {
+            adSourceId = jsonHeaders.getString(AD_SOURCE_ID_KEY);
```

Comment:
could it use that extractHeader() method like code above? just for better consistency

Reply:
Good call, fixed.